### PR TITLE
Opgops 2100

### PR DESCRIPTION
**Pull request by tim-mcarthur-moj**

### CODE REVIEW

File: cotton/salt/tasks.py [hunk 1]
```diff
@@ -54,17 +57,16 @@ def highstate_complete():
...
 
     while len(result.getvalue().strip()):
         result.truncate(0)
-        print(yellow("Highstate is still running.\nPolling again in {} seconds.\n".format(timeout)))
+        print yellow("Highstate is still running.\nPolling again in {} seconds.\n".format(timeout))
         time.sleep(timeout)
         salt_run(method='jobs.active', stdout=result)
 
-    print(green("Highstate complete.\n"))
+    print green("Highstate complete.\n")
```
Comment (Azulinho) [inline cotton/salt/tasks.py]: keeping the ( )  will keep it py3 friendly. even if the rest of the code is not

Comment (Azulinho) [inline cotton/salt/tasks.py]: on L69 and L65

File: cotton/salt/tasks.py [hunk 2]
```diff
@@ -192,7 +210,7 @@ def __update_master_config(keys):
 
     # read master config into dict
     master_data = StringIO()
-    sudo ('cat /etc/salt/master|grep -v ^# | grep -v ^$', stdout=master_data)
+    sudo('cat /etc/salt/master|grep -v ^# | grep -v ^$', stdout=master_data)
     salt_output = master_data.getvalue().strip()
```
Comment (Azulinho) [inline cotton/salt/tasks.py]: L195 how did this ever worked?

File: cotton/salt/tasks.py [hunk 3]
```diff
@@ -202,35 +220,39 @@ def __update_master_config(keys):
     # master_config = safe_load(master_data.getvalue())
     current_pillar_roots = master_config['pillar_roots'] if 'pillar_roots' in master_config else []
     new_pillar_roots = keys
+    if target is not None and target not in new_pillar_roots:
+        print "[WARN] Invalid target environment: {} \n Not updating salt configuration".format(target)
+        return True
 
-    if current_pillar_roots != new_pillar_roots:
+    if target is not None and target not in current_pillar_roots:
```
Comment (Azulinho) [inline cotton/salt/tasks.py]: how about ```if target and target not in current_pillar_roots: ```

File: cotton/salt/tasks.py [hunk 4]
```diff
@@ -202,35 +220,39 @@ def __update_master_config(keys):
...
-            f = open(d + '/master.auto', 'w')
-            f.write(safe_dump(master_config, default_flow_style=False))
-            f.close()
-            put(f.name, 'master.auto')
-            local('rm -rf ' + d)
+            tmp_dir = tempfile.mkdtemp()
+            master_config_file = '{}/master.auto'.format(tmp_dir)
+            master_file = open(master_config_file, 'w')
+            master_file.write(safe_dump(master_config, default_flow_style=False))
+            master_file.close()
```
Comment (Azulinho) [inline cotton/salt/tasks.py]: use a with block?
```
with open(master_config_file, "w") as f:
    f.write(safe_dump(master_config, default_flow_style=False))
```

File: cotton/salt/tasks.py [hunk 5]
```diff
@@ -202,35 +220,39 @@ def __update_master_config(keys):
...
+            put(master_file.name, 'master.auto')
+            local('rm -rf {}'.format(tmp_dir))
             sudo('mv  -v master.auto /etc/salt/master', stdout=master_data)
-            print("{}".format(master_data.getvalue()))
+            print "{}".format(master_data.getvalue())
```
Comment (Azulinho) [inline cotton/salt/tasks.py]: same comment as above, print() is py3 friendly too

File: cotton/salt/tasks.py [hunk 6]
```diff
@@ -202,35 +220,39 @@ def __update_master_config(keys):
...
-            pass
+        except Exception as err:
+            print str(err)
     elif refresh:
-        print('salt needs to have the pillar cache rebuilt')
-        reload_pillar()
-    # TODO run a salt reload cache here for changed pillar data
-    #
+        print 'Refreshing pillar cache'
+        if target is not None:
```
Comment (Azulinho) [inline cotton/salt/tasks.py]: maybe just ```if target:``` ?

**Comment by carinadigital** [comment]
Very little code has changed. It's mainly code formatting and a set of functions passing `target` down.
Happy with the changes

File: cotton/salt/tasks.py (hunk 1 above)
Comment (Azulinho) [inline cotton/salt/tasks.py]: yeap, you can disable those  print-statement (E1601)
but not really worth spending more time on this

**Comment by Azulinho** [comment]
looks everyone signed it off.
thanks for fixing it

**Pull request merged**